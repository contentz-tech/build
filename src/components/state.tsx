import { createContext, useContext } from "react";
import { jsx } from "@emotion/core";
import { IState } from "../state";

const StateContext = createContext<IState | null>(null);

interface StateProviderProps {
  state: IState;
  children: JSX.Element[] | JSX.Element;
}

function StateProvider({ state, children }: StateProviderProps) {
  return (
    <StateContext.Provider value={state}>{children}</StateContext.Provider>
  );
}

function useState(): IState {
  const state = useContext(StateContext);
  if (state === null)
    throw new Error(
      "Wrap your component in `StateProvider` and pass a valid state."
    );
  return state;
}

export { StateProvider, useState };
