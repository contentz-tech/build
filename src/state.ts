import { IConfig, config as getConfig } from "./getters/config";
import { articles, IArticles } from "./getters/articles";
import { pages, IPages } from "./getters/pages";
import { slides, ISlides } from "./getters/slides";
import { links, ILink } from "./getters/links";
import { statics } from "./getters/statics";
import { i18n } from "./getters/i18n";

interface IState {
  config: IConfig;
  articles: IArticles;
  pages: IPages;
  slides: ISlides;
  links: ILink[];
  statics: string[];
  i18n: any;
}

async function getState(): Promise<IState> {
  const config: IConfig = await getConfig();
  return {
    config,
    articles: await articles(config),
    pages: await pages(config),
    slides: await slides(config),
    links: await links(),
    statics: await statics(),
    i18n: await i18n(config)
  };
}

export { getState, IState };
