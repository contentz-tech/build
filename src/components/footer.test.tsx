/** * @jest-environment jsdom */
import "@testing-library/jest-dom/extend-expect";
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import Footer from "./footer";
import { StateProvider } from "./state";

const baseState = {
  config: { language: "en" },
  articles: {
    order: [],
    byPath: {}
  },
  slides: {
    order: [],
    byPath: {}
  },
  statics: [],
  pages: {
    order: [],
    byPath: {}
  },
  links: [],
  i18n: {
    footer: { editOnGitHub: "Edit it on GitHub" },
    patreon: {
      first: "Do you like my content?",
      link: "Become a Patreon and help me continue writing!"
    }
  }
};

describe("Footer", () => {
  test("Empty", () => {
    const { asFragment } = render(
      <StateProvider state={baseState}>
        <Footer />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Only Repository", () => {
    const { asFragment, getByText } = render(
      <StateProvider
        state={{
          ...baseState,
          config: {
            ...baseState.config,
            repository: "https://github.com/contentz-tech/build"
          }
        }}
      >
        <Footer repositoryPath="https:/github.com/contentz-tech/build/edit/master/src/components/footer.test.tsx" />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(baseState.i18n.footer.editOnGitHub)).toBeInTheDocument();
    expect(getByText(baseState.i18n.footer.editOnGitHub)).toHaveAttribute(
      "rel",
      "alternate"
    );
    expect(getByText(baseState.i18n.footer.editOnGitHub)).toHaveAttribute(
      "title",
      baseState.i18n.footer.editOnGitHub
    );
    expect(getByText(baseState.i18n.footer.editOnGitHub)).toHaveAttribute(
      "href",
      "https:/github.com/contentz-tech/build/edit/master/src/components/footer.test.tsx"
    );
  });

  test("With Patreon", () => {
    const { asFragment, getByText } = render(
      <StateProvider state={{ ...baseState, config: { patreon: "sergiodxa" } }}>
        <Footer />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/Do you like my content?/)).toBeInTheDocument();
    expect(
      getByText(/Become a Patreon and help me continue writing!/)
    ).toBeInTheDocument();
    expect(
      getByText(/Become a Patreon and help me continue writing!/)
    ).toHaveAttribute("href", "https://patreon.com/sergiodxa");
  });

  test("Complete", () => {
    const { asFragment, getByText } = render(
      <StateProvider
        state={{
          ...baseState,
          config: {
            patreon: "sergiodxa",
            repository: "https://github.com/contentz-tech/build"
          }
        }}
      >
        <Footer repositoryPath="https:/github.com/contentz-tech/build/edit/master/src/components/footer.test.tsx" />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(baseState.i18n.footer.editOnGitHub)).toBeInTheDocument();
    expect(getByText(baseState.i18n.footer.editOnGitHub)).toHaveAttribute(
      "rel",
      "alternate"
    );
    expect(getByText(baseState.i18n.footer.editOnGitHub)).toHaveAttribute(
      "title",
      baseState.i18n.footer.editOnGitHub
    );
    expect(getByText(baseState.i18n.footer.editOnGitHub)).toHaveAttribute(
      "href",
      "https:/github.com/contentz-tech/build/edit/master/src/components/footer.test.tsx"
    );

    expect(getByText(/Do you like my content?/)).toBeInTheDocument();
    expect(
      getByText(/Become a Patreon and help me continue writing!/)
    ).toBeInTheDocument();
    expect(
      getByText(/Become a Patreon and help me continue writing!/)
    ).toHaveAttribute("href", "https://patreon.com/sergiodxa");
  });
});
