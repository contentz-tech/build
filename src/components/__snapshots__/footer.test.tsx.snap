// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Complete 1`] = `
<DocumentFragment>
  .emotion-6 {
  background: white;
  border-top: 1px solid #eaeaea;
  padding: 2em 0;
  margin-top: 3em;
}

.emotion-5 {
  max-width: 60rem;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (max-width:60rem) and (orientation:portrait) {
  .emotion-5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 0 2em;
  }
}

@media (max-width:60rem) and (orientation:landscape) {
  .emotion-5 {
    padding: 0 2em;
  }
}

.emotion-4 {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.8em;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:60rem) {
  .emotion-4 {
    margin-top: 1em;
  }
}

.emotion-3 {
  height: 18px;
  width: 18px;
  margin-left: 10px;
}

.emotion-2 {
  fontsize: 0.9em;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: #004BCC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<footer
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-1"
        >
          Do you like my content? 
          <br />
           
          <a
            class="emotion-0"
            href="https://patreon.com/sergiodxa"
            rel="nofollow noopener"
            target="_blank"
          >
            Become a Patreon and help me continue writing!
          </a>
        </p>
      </div>
      <a
        class="emotion-4"
        href="https:/github.com/contentz-tech/build/edit/master/src/components/footer.test.tsx"
        rel="alternate"
        title="Edit it on GitHub"
      >
        Edit it on GitHub 
        <i
          class="emotion-3"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
            />
          </svg>
        </i>
      </a>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer Empty 1`] = `<DocumentFragment />`;

exports[`Footer Only Repository 1`] = `
<DocumentFragment>
  .emotion-3 {
  background: white;
  border-top: 1px solid #eaeaea;
  padding: 2em 0;
  margin-top: 3em;
}

.emotion-2 {
  max-width: 60rem;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (max-width:60rem) and (orientation:portrait) {
  .emotion-2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 0 2em;
  }
}

@media (max-width:60rem) and (orientation:landscape) {
  .emotion-2 {
    padding: 0 2em;
  }
}

.emotion-1 {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.8em;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:60rem) {
  .emotion-1 {
    margin-top: 1em;
  }
}

.emotion-0 {
  height: 18px;
  width: 18px;
  margin-left: 10px;
}

<footer
    class="emotion-3"
  >
    <div
      class="emotion-2"
    >
      <a
        class="emotion-1"
        href="https:/github.com/contentz-tech/build/edit/master/src/components/footer.test.tsx"
        rel="alternate"
        title="Edit it on GitHub"
      >
        Edit it on GitHub 
        <i
          class="emotion-0"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
            />
          </svg>
        </i>
      </a>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer With Patreon 1`] = `
<DocumentFragment>
  .emotion-4 {
  background: white;
  border-top: 1px solid #eaeaea;
  padding: 2em 0;
  margin-top: 3em;
}

.emotion-3 {
  max-width: 60rem;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (max-width:60rem) and (orientation:portrait) {
  .emotion-3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 0 2em;
  }
}

@media (max-width:60rem) and (orientation:landscape) {
  .emotion-3 {
    padding: 0 2em;
  }
}

.emotion-2 {
  fontsize: 0.9em;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: #004BCC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<footer
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-1"
        >
          Do you like my content? 
          <br />
           
          <a
            class="emotion-0"
            href="https://patreon.com/sergiodxa"
            rel="nofollow noopener"
            target="_blank"
          >
            Become a Patreon and help me continue writing!
          </a>
        </p>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
